include/libtorrent/storage.hpp

```diff
@@ -689,6 +689,8 @@ namespace libtorrent
 		// the piece_manager destructs. This is because
 		// the torrent_info object is owned by the torrent.
 		boost::shared_ptr<void> m_torrent;
+
+		bool checked_fastresume;
```

Comment:
I don't really understand what this solves. When a user force-recheck a torrent, do we come back through the check_fastresume() case? In either case, I think this would be a good place to describe what this variable is used for and why it's needed